File: app/interactors/webhooks/find_fixable_issues.rb
module Webhooks
  class FindFixableIssues
    include Interactor
    include InteractorHelper

    def_delegators :context, :body

    def call
      matches = body.scan(/(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s*(\#\d+|http.*\/\d+)/i)

      context.numbers = matches.map!(&:last).map! do |issue|
        if issue.start_with?('#') # #7631"
          issue.sub('#', '')
        else
          issue.split('/').last # https://github.com/woodmont/spreeworks/issues/7631
        end
      end.uniq
    end
  end
end

File: lib/findFixableIssues.js
module.exports = body => {
  const matches = body.match(
    /(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s*(#\d+|http.*\/issues\/\d+)/gi
  )

  return Array.from(
    new Set(
      matches.map(match => {
        if (match.indexOf('#') > -1) {
          // #1234
          return match.split('#').pop()
        } else {
          // https://github.com/org/repo/issues/1234
          return match.split('/').pop()
        }
      })
    )
  )
}
